lib/pbench/test/unit/agent/test_utils.py

```diff
@@ -97,17 +97,21 @@ def test_kill(self, tmp_path, monkeypatch):
         with pytest.raises(AssertionError):
             bs.kill(1)
 
-        bs = OurServer("localhost", "localhost")
         pidfile = tmp_path / "test.pid"
+
+        bs = OurServer("localhost", "localhost")
         pidfile.write_text("12345")
         bs.pid_file = pidfile
         ret = bs.kill(42)
         assert ret == 42
```

Comment:
Shouldn't we be mocking `OurServer.get_pid()` (or `BaseServer.get_pid()`), `os.kill()`, and possibly the `time` functions?

Reply:
Yeah, well, I wouldn't even have touched this file except that I added that excessive pid caching, which broke the test as it re-used the `OurServer` object instance while changing `pid_file`. For some reason I just couldn't make myself go back to drop the `pid` caching, though, so I had to either add `bs.pid = None` (which I didn't like) or use a fresh `OurServer` for each case.

But you're right; mocking `os.kill` would have been good. I don't think I would want to mock `BaseServer.get_pid()`, however mocking the `Path.read_text` would avoid using a real file.

All things to consider in refactoring and improving the unit tests....

Reply:
> I wouldn't even have touched this file except that I added that excessive pid caching
 😀

> I don't think I would want to mock `BaseServer.get_pid()`

That depends on how you define the "unit" and how you want to inject the PID value into the CUT (i.e., you _could_ use a `Path.read_text()` mock for that, or you could mock out the _need_ for a `Path.read_text()` mock...).

> All things to consider in refactoring and improving the unit tests....

Yay!  Thank you....